syntax = "proto3";

package automatedcodingworkflow;

service AutomatedCodingWorkflowService {
    rpc StartWorkflow (StartWorkflowRequest) returns (StartWorkflowResponse);
    rpc GetWorkflowConfig (GetWorkflowConfigRequest) returns (GetWorkflowConfigResponse);
    rpc SetWorkspacePath (SetWorkspacePathRequest) returns (SetWorkspacePathResponse);
}

message StartWorkflowRequest {
}

message StartWorkflowResponse {
    string result = 1;
}

message GetWorkflowConfigRequest {
}

message GetWorkflowConfigResponse {
    repeated Step steps = 1;
}

message Step {
    string step_name = 1;
    string step_class = 2;
    repeated Step steps = 3;
}

message SetWorkspacePathRequest {
    string workspace_path = 1;
}
  
message SetWorkspacePathResponse {
    bool success = 1;
    string error_message = 2;
}
